docs/HARDWARE-MODEL.md

```diff
@@ -0,0 +1,154 @@
+
+## Model description files
+
+The model files are kept in [eden/models](https://github.com/lf-edge/eden/tree/master/models) and are composed of several parts:
+
+- General description including a product URL
+- Attributes for CPU cores, memory, storage, TPM, etc
+- Front and back images
+- I/O adapters
+
+### Product URL
+
+This is supposed to be a URL which identifies the model in sufficient detail that someone else can order it and EVE will run the same.
+
+### Front and back images
+
+The purposes of these images is to create a binding between the physical ports on the device and the names used in the model file. In some cases the device enclosure has unique names printed next to each port, and in other cases it is useful to annotate the image with unique names. And example of the former is [logo_nacl_MOXA-1200](https://github.com/lf-edge/eden/blob/master/models/logo_back_MOXA-1200.jpg) and an example of the latter is [logo_back_MBI-6418A-T7H](https://github.com/lf-edge/eden/blob/master/models/logo_back_MBI-6418A-T7H.jpg).
+
+### I/O Adapters
+
+The purpose of the I/O adapters section is to describe the networking ports which can be used by EVE-OS to reach the controller, and also to describe networking, other I/O ports, and GPUs which can potentially be assigned to application instances.
+
+Serial ports can be assigned to app instances without much restrictions; in some cases an IRQ might be shared between a COM3 and COM4 port. For security and stability reasons the Xen and KVM hypervisors restrict the assignment of PCI devices and controllers based on the capabilities of the IOMMU. This is to prevent one application instance to be able to program a DMA engine in an I/O device to use DMA to read and/or write from other parts of the physical memory in the device. (This is captured using a concept called IOMMU groups.)
+
+The model file describes these assignment constraints using the assigngrp field. If the assigngrp field is empty it means that the adapter can not be assigned to application instances (but if it is a networking adapter it can still be used by EVE-OS). Otherwise all of the adapters with the same assignment group can be assigned as a unit to the same application instance. An example of this is a GPU device which might have a graphics and a separate audio device, which are in the same group.
+
+USB controllers vs USB ports vs cellular modems requires some additional work and might place some more constraints as specified below.
+
+Last but not least, there are reports in industry that for some devices (in particular GPUs) the IOMMU group specification might be incorrect. Thus it is critical that the model description is tested by doing actual assignment to application instances.
+
+## Generating initial model file
+
+When EVE-OS is booted on a device one can run a script to get a partial and initial model file. That script is invoked as:
```

Comment:
I REALLY want to make sure we wordsmith it somehow to make sure everybody understands that this capability is only available in a "test" or "developer" image. I really would like to stay away from promoting the notion that you can "ssh into the EVE node"

Reply:
Updated